frida_mode/src/cmplog/cmplog_x64.c

```diff
@@ -247,13 +254,8 @@ static void cmplog_instrument_cmp_sub(const cs_insn *     instr,
   if (operand1->type == X86_OP_INVALID) return;
   if (operand2->type == X86_OP_INVALID) return;
 
-  if ((operand1->type == X86_OP_MEM) &&
-      (operand1->mem.segment != X86_REG_INVALID))
-    return;
-
-  if ((operand2->type == X86_OP_MEM) &&
-      (operand2->mem.segment != X86_REG_INVALID))
-    return;
+  /* Both operands are the same size */
+  if (operand1->size == 1) { return; }
```

Comment:
can that even happen? I mean there is a cmp. so why ignore it?

Reply:
Single byte compares/subtracts are a thing. But QEMU mode ignores them...
https://github.com/AFLplusplus/qemuafl/blob/d73b0336b451fd034e5f469089fb7ee96c80adf2/qemuafl/cpu-translate.h#L88
Presumably, this is because the overhead of cmplog is not worthwhile compared to just letting the fuzzer guess the byte the normal way.

Reply:
you are right